shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1396,12 +1430,40 @@ - (void)setupKeyboardAnimationVsyncClient {
       // Ensure the keyboardAnimationView is in view hierarchy when animation running.
       [flutterViewController.get().view addSubview:[flutterViewController keyboardAnimationView]];
     }
-    if ([flutterViewController keyboardAnimationView].layer.presentationLayer) {
-      CGFloat value =
-          [flutterViewController keyboardAnimationView].layer.presentationLayer.frame.origin.y;
-      flutterViewController.get()->_viewportMetrics.physical_view_inset_bottom = value;
-      [flutterViewController updateViewportMetrics];
+
+    CGFloat newY;
+    if ([flutterViewController keyboardSpringCurve] == nil) {
+      newY = [flutterViewController keyboardAnimationView].layer.presentationLayer.frame.origin.y;
+    } else {
+      fml::TimeDelta timeElapsed =
+          recorder.get()->GetVsyncTargetTime() - [flutterViewController keyboardAnimationStartTime];
+      double maxFrameRate = [DisplayLinkManager displayRefreshRate];
+      double frameRate = [[flutterViewController keyboardAnimationVSyncClient] getRefreshRate];
+
+      double animationSettlingDuration = [flutterViewController keyboardAnimationDuration];
+      double keyboardAnimationTimeMillis = animationSettlingDuration * 1000;
+
+      double expectedFrames = frameRate * animationSettlingDuration * (maxFrameRate / frameRate);
+      double percentComplete = timeElapsed.ToMillisecondsF() / keyboardAnimationTimeMillis;
+
+      int frameApproximation = roundf(MIN(percentComplete, 1) * expectedFrames);
+      double frameTime = frameApproximation / maxFrameRate;
+
+      double keyboardAnimationStop;
+      // Ensure the animation finishes with 100%
+      if (frameApproximation == expectedFrames) {
+        keyboardAnimationStop = 1;
+      } else {
+        keyboardAnimationStop = [[flutterViewController keyboardSpringCurve] curveFunc:frameTime];
+      }
+
+      newY = [flutterViewController keyboardAnimationFrom] +
+             keyboardAnimationStop * ([flutterViewController keyboardAnimationTo] -
+                                      [flutterViewController keyboardAnimationFrom]);
```

Comment:
I think maybe this algorithm in `else` branch can be replace with brief code just:
```objc
  double start = flutterViewController.get().keyboardAnimationFrom;
  double end = flutterViewController.get().keyboardAnimationTo;
  fml::TimeDelta timeElapsed =
    recorder.get()->GetVsyncTargetTime() - flutterViewController.get().keyboardAnimationStartTime;
  double keyboardAnimationStop = [[flutterViewController keyboardSpringCurve] curveFunc:timeElapsed.ToSecondsF()];
  newY = start + (end - start) * keyboardAnimationStop;
```

Reply:
Haha funny how much simpler things can be! Thanks :)